Fails to run in Cloudflare Workers

```bash
✘ [ERROR] A request to the Cloudflare API failed.

  Uncaught Error: EvalError: Code generation from strings disallowed for
  this context
    at null.<anonymous>
  (file:///home/kyle/projects/coder/blink/worker/node_modules/memium/dist/struct.js:52:24)
  in __decorateStruct
    at null.<anonymous>
  (file:///home/kyle/projects/coder/blink/worker/node_modules/@zenfs/core/dist/internal/inode.js:12:38)
  in __esDecorate
    at null.<anonymous>
  (file:///home/kyle/projects/coder/blink/worker/node_modules/@zenfs/core/dist/internal/inode.js:128:9)
    at null.<anonymous>
  (file:///home/kyle/projects/coder/blink/worker/node_modules/@zenfs/core/dist/internal/inode.js:132:5)
    at null.<anonymous>
  (file:///home/kyle/projects/coder/blink/worker/node_modules/@zenfs/core/dist/internal/inode.js:134:1)
   [code: 10021]
```

I tried to remove the eval, which is here:
```ts
        // This is so we preserve the name of the class
        const struct = new Function('target', `return class ${target.name} extends target {
				constructor(...args) {
					if (!args.length) args = [new ArrayBuffer(${size}), 0, ${size}];
					super(...args);
				}
			}`)(target);
```
It caused a whole swarm of other issues to appear. Any help would be appreciated!

--- a/src/struct.ts
+++ b/src/struct.ts
@@ -63,15 +63,15 @@ export function struct(...options: Options[]) {
 		} satisfies Metadata;
 
 		// This is so we preserve the name of the class
-		const struct = new Function(
-			'target',
-			`return class ${target.name} extends target {
-				constructor(...args) {
-					if (!args.length) args = [new ArrayBuffer(${size}), 0, ${size}];
-					super(...args);
-				}
-			}`
-		)(target);
+		// @ts-expect-error
+		const struct = class extends target {
+			public static readonly name = target.name;
+
+			constructor(...args: any[]) {
+				if (!args.length) args = [new ArrayBuffer(size), 0, size];
+				super(...args);
+			}
+		};
 
 		const fix = (value: any) => ({
 			writable: false,
@@ -91,7 +91,7 @@ export function struct(...options: Options[]) {
 			}),
 		});
 
-		registerType(struct);
+		registerType(struct as unknown as Type<InstanceType<T>>);
 
 		return struct;
 	};

